src/query.c

```diff
@@ -443,6 +443,32 @@ static void QueryNode_Expand(RSQueryTokenExpander expander, RSQueryExpanderCtx *
     return;
   }
 
+  // Check that there is at least one stemmable field in the query
+  if (expCtx->handle && expCtx->handle->spec) {
+    const IndexSpec *spec = expCtx->handle->spec;
+    t_fieldMask fm = qn->opts.fieldMask;
+    if ( fm != RS_FIELDMASK_ALL) {
+      int expand = 0;
+      t_fieldMask bit_mask = 1;
+      while (fm) {
+        if (fm & bit_mask) {
+          const FieldSpec *fs = IndexSpec_GetFieldByBit(spec, bit_mask);
+          if (fs) {
+            if(!FieldSpec_IsNoStem(fs)) {
+              expand = 1;
+              break;
+            }
+          }
```

Comment:
Consider a shorter format
```suggestion
          if (fs && !FieldSpec_IsNoStem(fs)) {
            expand = 1;
            break;
          }
```